### PR TITLE
feat: [SDKCF-3199] add close campaign API and refactor some test codes

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Description
This new API is for programmatically closing campaigns without user interaction.

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: inappmessaging/USERGUIDE.md [hunk 1]
```diff
@@ -218,6 +218,17 @@ InAppMessaging.instance().onVerifyContext = { contexts: List<String>, campaignTi
 }
 ```
 
+### <a name="close-campaign"></a> #2 Close campaign API
+There are cases that apps need to manually close the campaigns without user interaction.
+
+An example is when a different user logs-in and the currently displayed campaign does not target the new user.
+
+It is possible that the new user did not close the campaign (tapping the 'X' button) when logging in. The app can force-close the campaign by calling the `closeMessage()` API.
```
Comment (imaginaris) [inline inappmessaging/USERGUIDE.md:226]: Is it worth adding an info that closed campaign message will not be counted as displayed?

**Review by imaginaris — APPROVED** [review]

File: inappmessaging/USERGUIDE.md (hunk 1 above)
Comment (vincentdelmundo) [reply inappmessaging/USERGUIDE.md:226]: Yes, it is better to add.
Added, please check.

**Review by imaginaris — APPROVED** [review]

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InAppMessaging.kt [hunk 2]
```diff
@@ -74,6 +74,13 @@ abstract class InAppMessaging internal constructor() {
     @RestrictTo(RestrictTo.Scope.LIBRARY)
     internal abstract fun getHostAppContext(): Context?
 
+    /**
+     * This methods manually closes the currently displayed message.
```
Comment (donnie-jp) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InAppMessaging.kt:78]: "This method manually" is superfluous. Just
>  Close the currently displayed message.

File: inappmessaging/USERGUIDE.md [hunk 3]
```diff
@@ -218,6 +218,19 @@ InAppMessaging.instance().onVerifyContext = { contexts: List<String>, campaignTi
 }
 ```
 
+### <a name="close-campaign"></a> #2 Close campaign API
+There are cases that apps need to manually close the campaigns without user interaction.
```
Comment (donnie-jp) [inline inappmessaging/USERGUIDE.md:222]: There are cases that apps -> There may be cases where apps

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InAppMessaging.kt (hunk 2 above)
Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InAppMessaging.kt:78]: Updated to `Close the currently displayed message.`

File: inappmessaging/USERGUIDE.md (hunk 3 above)
Comment (vincentdelmundo) [reply inappmessaging/USERGUIDE.md:222]: Changed to the suggested phrase.

**Review by MdUsmanGani — APPROVED** [review]
LGTM

**Pull request merged**